### PR TITLE
New login

### PR DESCRIPTION
**Pull request by avaer**

### CODE REVIEW

**Comment by ChrisEddy** [comment]
too much code to know if it works but syntax looks good

File: login.js [hunk 1]
```diff
@@ -1,28 +1,67 @@
...
+      limit: 100,
+      script: contractSource.replace(/ARG0/g, '0x' + loginToken.addr),
+      wait: true,
+    }),
+  });
+  const response = await res.json();
+  const isBaked = response.encodedData.value;
+  if (!isBaked) {
+    const contractSources = await getContractSource('bakeUserAccount.json');
+    for (const contractSource of contractSources) {
```
Comment (ChrisEddy) [inline login.js:27]: contractSources can be null

File: login.js [hunk 2]
```diff
@@ -1,28 +1,67 @@
...
+    method: 'POST',
+    body: JSON.stringify({
+      limit: 100,
+      script: contractSource.replace(/ARG0/g, '0x' + loginToken.addr),
+      wait: true,
+    }),
+  });
+  const response = await res.json();
+  const name = response.encodedData.value[0].value ? response.encodedData.value[0].value.value : 'Anonymous';
+  const avatarHash = response.encodedData.value[1].value && response.encodedData.value[1].value.value;
```
Comment (ChrisEddy) [inline login.js:58]: value[1].value.value pretty weird

File: login.js [hunk 3]
```diff
@@ -32,17 +71,31 @@ async function pushUserObject() {
...
+async function finishLogin(newLoginToken) {
+  await storage.set('loginToken', newLoginToken);
+
+  loginToken = newLoginToken;
+
+  console.log('got user token', loginToken);
+
+  const loginForm = document.getElementById('login-form');
+  // document.body.classList.add('logged-in');
+  loginForm.classList.remove('phase-1');
```
Comment (ChrisEddy) [inline login.js:92]: To do: not a huge fan of the phase system. It works tho for now. Preact can handle this with some simple props now.

File: login.js [hunk 4]
```diff
@@ -147,27 +167,57 @@ async function tryLogin() {
...
+  /* document.getElementById('unloadAvatar').addEventListener('click', async (e) => {
     e.preventDefault();
     e.stopPropagation();
     loginManager.setAvatar(null);
-  })
+  }); */
+
+  const userName = document.getElementById('user-name');
+  userName.addEventListener('click', e => {
+    userName.setAttribute('contenteditable', '');
```
Comment (ChrisEddy) [inline login.js:180]: All this can be preact soon.

**Review by ChrisEddy — APPROVED** [review]

File: login.js (hunk 1 above)
Comment (avaer) [reply login.js:27]: That would be a serious error in the contracts deployment that we probably can't recover from anyway, so I'm fine with that crashing here.

File: login.js (hunk 2 above)
Comment (avaer) [reply login.js:58]: Yeah. Unfortunately I don't know a better way to peel the data off, though we might be able to abstract these more in the future.

File: login.js (hunk 3 above)
Comment (avaer) [reply login.js:92]: I agree, though that is not related to this PR.

**Pull request merged**